### PR TITLE
Move tests from WCServerTest that update server.xml to their own classes

### LINKED ISSUE #29524: Servlet 3.1 FAT: Move tests from `WCServerTest` that update server.xml to new classes
Servlet 3.1 FAT: Move tests from `WCServerTest` that update server.xml to new classes to avoid applications not being available after they run. There are a number of applications used in `WCServerTest` so cleaner to just move these few tests to new test classes.

### PR DESCRIPTION
**Pull request by pnicolucci**
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

fixes #29524

### CODE REVIEW

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/DecodeUrlPlusSignTests.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2024 IBM Corporation and others.
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/DecodeUrlPlusSignTests.java:2]: originally done: https://github.com/OpenLiberty/open-liberty/pull/12085 so using the 2020 copyright date as the base

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/ServletMappingTests.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
+/*******************************************************************************
+ * Copyright (c) 2014, 2024 IBM Corporation and others.
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/ServletMappingTests.java:2]: Added originally in : https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.workitem.viewWorkItem&id=131083 so using 2014 as the original copyright date

File: dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/WCServerTest.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2016, 2024 IBM Corporation and others.
+ * Copyright (c) 2013, 2024 IBM Corporation and others.
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer.servlet.3.1_fat/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/WCServerTest.java:2]: When this was ported: https://github.com/OpenLiberty/open-liberty/pull/11642 it should have been 2013 according to the previous history here: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.scm.browseElement&workspaceItemId=_YLyM4EWoEeWjDtfHC2Pi5w&componentItemId=_KjUsAAFkEeCRyLCwaAjryg&itemType=com.ibm.team.filesystem.FileItem&itemId=_HGAsMOr4EeKtAagKu-hDVA&tab=history

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#build

**Comment by pnicolucci** [comment]
#libby

**Comment by pnicolucci** [comment]
#build

**Review by pmd1nh — APPROVED** [review]

**Pull request merged**